script.js

```diff
@@ -318,3 +319,37 @@ form.addEventListener("submit", (e) => {
       "Please, use only lowercase letter for your email!";
   }
 });
+
+window.addEventListener('load', () => {
+  if (localStorage.getItem('userInfo')) {
+    const userInfo = JSON.parse(localStorage.getItem('userInfo'));
+    usernameInput.value = userInfo.name;
+    emailInput.value = userInfo.email;
+    messageText.value = userInfo.message;
+  } else {
+    localStorage.setItem('userInfo', userData);
+  }
+});
+
+function saveData(key, value) {
+  if (localStorage.getItem('userInfo')) {
+    const oldData = JSON.parse(localStorage.getItem('userInfo'));
+    const newData = {
+      ...oldData,
+      [key]: value
+    };
+    localStorage.setItem('userInfo', JSON.stringify(newData));
+  } else {
+    localStorage.setItem('userInfo', userData);
+  }
+}
+
+form.addEventListener('input', (evt) => {
+  if (evt.target.dataset.id === 'name' || evt.target.dataset.id === 'email' || evt.target.dataset.id === 'message') {
+    saveData(evt.target.dataset.id, evt.target.value);
+  }
+});
+
```

Comment:
* When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data, currently your form doesn't do this please fix this as its part of the project requirement

Reply:
ok corrected it